raiden/routing.py

```diff
@@ -129,3 +163,49 @@ def get_best_routes(
         route_state = RouteState(partner_address, channel_state_id)
         available_routes.append(route_state)
     return available_routes
+
+
+def get_best_routes_pfs(
+        chain_state: ChainState,
+        token_network_id: typing.TokenNetworkID,
+        from_address: typing.InitiatorAddress,
+        to_address: typing.TargetAddress,
+        amount: int,
+        previous_address: typing.Optional[typing.Address],  # TODO(paul): Remove this?
+        config: Dict,
+) -> List[RouteState]:
+    pfs_path = '{}/api/v1/{}/paths'.format(
+        config['pathfinding_service_address'],
+        to_checksum_address(token_network_id),
+    )
+    payload = {
+        'from': to_checksum_address(from_address),
+        'to': to_checksum_address(to_address),
+        'value': amount,
+        'max_paths': config['pathfinding_max_paths'],
+    }
+    response = requests.get(pfs_path, params=payload)
```

Comment:
A few things here:
- This needs error handling (the base class for all requests errors is `requests.RequestException`)
- It also needs a timeout. By default requests has no defined timeout, that means this request could hang indefinitely. 
- As an optimization we could use a [`Session`](http://docs.python-requests.org/en/master/user/advanced/#session-objects) here, that would automatically enable HTTP keep-alive

Reply:
Good points.

- Is there a default for a timeout in raiden?
- I'd need to save the session, probably in the `RaidenService` class. Do you think that's worth it?

Reply:
> I'd need to save the session, probably in the RaidenService class. Do you think that's worth it?

Yeah probably not. Premature optimization and all that ;)